extensions/blocks/tiled-gallery/utils/index.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * External dependencies
+ */
+import photon from 'photon';
+import { format as formatUrl, parse as parseUrl } from 'url';
+import { isBlobURL } from '@wordpress/blob';
+import { range } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { PHOTON_MAX_RESIZE } from '../constants';
+
+export function isSquareishLayout( layout ) {
+	return [ 'circle', 'square' ].includes( layout );
+}
+
+/**
+ * Build src and srcSet properties which can be used on an <img />
+ *
+ * @param {Object} img        Image
+ * @param {number} img.height Image height
+ * @param {string} img.url    Image URL
+ * @param {number} img.width  Image width
+ *
+ * @param {Object} galleryAtts Gallery attributes relevant for image optimization.
+ * @param {string} galleryAtts.layoutStyle Gallery layout. 'rectangular', 'circle', etc.
+ * @param {number} galleryAtts.columns     Gallery columns. Not applicable for all layouts.
+ *
+ * @return {Object} Returns an object. If possible, the object will include `src` and `srcSet`
+ *                  properties {string} for use on an image.
+ */
+export function photonizedImgProps( img, galleryAtts = {} ) {
+	if ( ! img.height || ! img.url || ! img.width ) {
+		return {};
+	}
+
+	// Do not Photonize images that are still uploading or from localhost
+	if ( isBlobURL( img.url ) || /^https?:\/\/localhost/.test( img.url ) ) {
+		return { src: img.url };
+	}
+
+	// Drop query args, photon URLs can't handle them
+	// This should be the "raw" url, we'll add dimensions later
+	const url = img.url.split( '?', 1 )[ 0 ];
+	const { height, width } = img;
+	const { layoutStyle } = galleryAtts;
+
+	const photonImplementation = isWpcomFilesUrl( url ) ? photonWpcomImage : photon;
+
+	/**
+	 * Build the `src`
+	 * We don't know what the viewport size will be like. Use full size src.
+	 */
+
+	let src;
+	if ( isSquareishLayout( layoutStyle ) && width && height ) {
+		// Layouts with 1:1 width/height ratio should be made square
+		const size = Math.min( PHOTON_MAX_RESIZE, width, height );
+		src = photonImplementation( url, {
+			resize: `${ size },${ size }`,
+			strip: 'all',
+		} );
+	} else {
+		src = photonImplementation( url, { strip: 'all' } );
+	}
+
+	/**
+	 * Build a sensible `srcSet` that will let the browser get an optimized image based on
+	 * viewport width.
+	 */
+
+	const step = 300;
+	const srcsetMinWith = 600;
+
+	let srcSet;
+	if ( isSquareishLayout( layoutStyle ) ) {
+		const minWidth = Math.min( srcsetMinWith, width, height );
+		const maxWidth = Math.min( PHOTON_MAX_RESIZE, width, height );
+
+		srcSet = range( minWidth, maxWidth, step )
+			.map( srcsetWidth => {
```

Comment:
BTW you're using a mixture of `srcset` and `srcSet` in variable names. Not a biggie thought _you_ ought to know. 😜

Reply:
😱 Yeah… I think I was mostly using `srcset`, but React wants `srcSet` so… 🤷‍♂️